.config/vite.config.js

```diff
@@ -26,13 +27,27 @@ export default defineConfig({
     tsconfigPaths({ loose: true }),
   ],
   build: {
+    target: "ES6",
```

Comment:
Why `ES6`? It's pretty old (2015). We'll be getting unnecessary polyfills (larger, less efficient bundles)

Our supported browsers seem to support at least [ES2020](https://caniuse.com/sr_es11) and [ES2021](https://caniuse.com/sr_es12) (didn't exhaustively search), so it should be safe to use `ES2021` here.

And if that's the case, then there's nothing to polyfill; because we're disallowing newer features statically with TS's `lib` `ES2021`

Reply:
I think we should keep `ES2017` if `ES2018` is not possible because of a browser version we say we support. Changing the browsers we support seems like something that should only happen on a major release.  

I also looked at the build time and it's around 2-3 min now in my machine.

Reply:
I think we need to be pragmatic here.

The problems that arose were due to usage of "bleeding-edge" (ES2022) features, not old ones. We're already not polyfilling to these older (ES2017/ES2018) versions and we haven't had issues regarding this.

By polyfilling everything back to ES2017/ES2018 we're generating extra code that:
- Takes longer to build
- Makes our bundles larger (longer downloads, loading times, etc.)
- Runtime isn't as efficient/optimised (nothing better than native, right?)

And what do we stand to gain for this? Compatibility with uncommon language features (eg. [lookbehind](https://caniuse.com/js-regexp-lookbehind)) that we don't use and probably never will? I don't think the trade-off make sense

Reply:
> I also looked at the build time and it's around 2-3 min now in my machine.

Build times aren't all that relevant, compared to the generated user code. I looked at the `ES6` transpiled code for the `HvButton`, and it almost doubled - with unnecessary polyfills, as browsers already supported it

Reply:
Okay, let's change it if you guys agree 👍 I don't have a strong opinion about this. I just wasn't sure about the browser compatibility change but, like you said, we are already not polyfilling to these versions.